src/Foundatio.RabbitMQ/Messaging/RabbitMQMessageBus.cs

```diff
@@ -117,21 +131,22 @@ public class RabbitMQMessageBus : MessageBusBase, IMessageBus {
                 return;
             _logger.Trace("Message Publish: {messageType}", messageType.FullName);
 
-            // RabbitMQ only supports delayed messages with a third party plugin.
-            if (delay.HasValue && delay.Value > TimeSpan.Zero) {
-                await AddDelayedMessageAsync(messageType, message, delay.Value).AnyContext();
-                return;
-            }
-
             var data = await _serializer.SerializeAsync(new MessageBusData {
                 Type = messageType.AssemblyQualifiedName,
                 Data = await _serializer.SerializeToStringAsync(message).AnyContext()
             }).AnyContext();
             
             var basicProperties = _publisherChannel.CreateBasicProperties();
             basicProperties.Persistent = _persistent;
-            basicProperties.Expiration = _defaultMessageTimeToLive.Milliseconds.ToString();
-
+            basicProperties.Expiration = _defaultMessageTimeToLive.TotalMilliseconds.ToString(CultureInfo.InvariantCulture);
+            // RabbitMQ only supports delayed messages with a third party plugin called "rabbitmq_delayed_message_exchange"
+            if (_delayedExchange && delay.HasValue && delay.Value > TimeSpan.Zero) {
```

Comment:
Well I tried to look for an api to find out if there is a way to know if plugin is installed but couldn't find one. But yes, if the plugin is not installed and you are trying to create x-delayed-message exchange then the exception gets thrown.

Unhandled Exception: RabbitMQ.Client.Exceptions.OperationInterruptedException: T
he AMQP operation was interrupted: AMQP close-reason, initiated by Peer, code=50
3, text="COMMAND_INVALID - invalid exchange type 'x-delayed-message'", classId=4
0, methodId=10, cause=
   at RabbitMQ.Client.Impl.SimpleBlockingRpcContinuation.GetReply(TimeSpan timeo
ut)
   at RabbitMQ.Client.Impl.ModelBase.ModelRpc(MethodBase method, ContentHeaderBa
se header, Byte[] body)
   at RabbitMQ.Client.Framing.Impl.Model._Private_ExchangeDeclare(String exchang
e, String type, Boolean passive, Boolean durable, Boolean autoDelete, Boolean in
ternal, Boolean nowait, IDictionary`2 arguments)
   at RabbitMQ.Client.Impl.ModelBase.ExchangeDeclare(String exchange, String typ
e, Boolean durable, Boolean autoDelete, IDictionary`2 arguments)
   at Foundatio.Messaging.RabbitMQMessageBus.SetUpExchangeAndQueuesForRouting(IM
odel model) in C:\Users\barkha.shah\Foundatio\src\Foundatio.RabbitMQ\Messaging\R
abbitMQMessageBus.cs:line 98
   at Foundatio.Messaging.RabbitMQMessageBus..ctor(String userName, String passw
ord, String queueName, String routingKey, String exhangeName, Boolean delayedExc
hange, Boolean durable, Boolean persistent, Boolean exclusive, Boolean autoDelet
e, IDictionary`2 queueArguments, Nullable`1 defaultMessageTimeToLive, ISerialize
r serializer, ILoggerFactory loggerFactory) in C:\Users\barkha.shah\Foundatio\sr
c\Foundatio.RabbitMQ\Messaging\RabbitMQMessageBus.cs:line 72
   at Foundatio.RabbitMQSubscribeConsole.Program.Main(String[] args)
Press any key to continue . . .